src/lib/utils/d3Graph.js

```diff
@@ -260,9 +261,43 @@ D3Graph.prototype.updateScale = function(options) {
     g.selectAll('.area')
       .transition().duration(TRANSITION_DURATION)
       .attr("d", this.area(temporalPoints))
+
+    g.selectAll('.rawDataPoints') // may be empty - that's ok!
+      .transition().duration(TRANSITION_DURATION)
+      .attr("cy", (d) => this.y(d.get(`raw_${this.params.key}`) || false))
   });
 }
 
+/**
+ * Prototype called when the "show raw data" toggle is changed.
+ */
+D3Graph.prototype.toggleRawDataPoints = function(options) {
+  if (this.params.graphType !== "lines") throw new Error("Not yet implemented")
+  if (!options.showRawData) {
+    this.svg.selectAll('.rawDataPoints').remove("*")
```

Comment:
_non-blocking_

I haven't tested the speed of the toggle, but in the measurements panel I sped up [toggling by updating the `display` attr](https://github.com/nextstrain/auspice/blob/3e3f8bd66af61d9aa082b50cc13e57c629d9a07e/src/components/measurements/measurementsD3.js#L411) instead of drawing/removing the raw data points.

Reply:
Yeah - that was a design I hadn't considered before you made the measurements panel. My thinking this time around was that people will rarely toggle raw data points on, so the speed penalty of having them in the DOM (but not displayed), and transitioning them when we go to logit etc wasn't worth it. But I haven't actually tested what penalty this would incur - do you have any ideas of the speed penalties / gains in these situations?

Reply:
Ah right, I hadn't considered the logit transitions 🤔 

I don't have exact numbers for speed gains, but it was a very obvious speed up for the measurements panel because there's significantly more raw data points. It made sense for the measurements panel to have an initial long load time then have the toggles be more responsive since the raw data values are static.